source/common/stats/thread_local_store.h

```diff
@@ -70,6 +70,7 @@ class ThreadLocalStoreImpl : public StoreRoot {
   void setTagExtractors(const std::vector<TagExtractorPtr>& tag_extractors) override {
     tag_extractors_ = &tag_extractors;
   }
+  void setDefaultTags(const std::vector<Tag>& tags) override { default_tags_ = &tags; }
```

Comment:
This seems risky. What's the lifespan of tags guaranteed to be? I think it is OK because tags has the lifespan of the server. But any time we are saving a pointer to a value passed by reference it's a good idea to document what controls the lifecycle of that memory.

Reply:
Yeah, a server hold the value so the lifespan will be same as the server's one. Looks good with this comment? a94dbda6b6048e6a99b0999e2ad6fe04f8bd84a9

Reply:
Finally stopped this approach and change thread local store to have ownership of those pointers in e0596988a490ae5a0ad78271523d2cc9046d6896.